dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -661,6 +663,78 @@ def test_sample_size_passed_to_profile(self, *mocks):
         profiler._sampling_ratio = 0.2
         self.assertEqual(10000, update_mock.call_args[0][1])
 
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnPrimitiveTypeProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnStatsProfileCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.'
+                'ColumnDataLabelerCompiler')
+    @mock.patch('dataprofiler.profilers.profile_builder.DataLabeler')
+    def test_index_overlap_resolved_for_update_profile(self, *mocks):
+        data = pd.Series([0, None, 1, 2, None])
+        profile = StructuredDataProfile(data)
+        self.assertEqual(0, profile.min_id)
+        self.assertEqual(4, profile.max_id)
+        self.assertDictEqual(profile.null_types_index, {'nan': {1, 4}})
+        profile.update_profile(data)
+        # Now all indices will be shifted by max_id + 1 (5)
+        # So the 2 None will move from indices 1, 4 to 6, 9
```

Comment:
I can see this works for the cases with time-series data, and the processing is over time. But in case that the user chooses randomly some proportions of the same large dataset, then profile them and merge profiles, does that work?